operator/pkg/utils/pki/pki.go

```diff
@@ -0,0 +1,196 @@
+/*
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package controller
+
+import (
+	"crypto"
+	cryptorand "crypto/rand"
+	"crypto/rsa"
+	"crypto/x509"
+	"crypto/x509/pkix"
+	"encoding/pem"
+	"errors"
+	"fmt"
+	"math"
+	"math/big"
+	"net"
+	"time"
+
+	"go.uber.org/zap"
+	"k8s.io/apimachinery/pkg/util/sets"
+	certutil "k8s.io/client-go/util/cert"
+	"k8s.io/client-go/util/keyutil"
+)
+
+const (
+	rsaKeySize          = 2048
+	CertificateValidity = time.Hour * 24 * 365
+)
+
+type CertConfig struct {
+	*certutil.Config
+	ExistingCert []byte
+	ExistingKey  []byte
+}
+
+// RootCA for a given config will check existing certs if they are valid, else
+// will generate new root CA for the certutil.Config provided
+func RootCA(config *CertConfig) (certBytes, keyBytes []byte, err error) {
```

Comment:
Thoughts on returning `type CertificateAuthority struct` instead of multiple return types?